### ISSUE TITLE
Thermal Expansion machine losing settings when upgrade from 2.1.0 to 2.2.0

### ISSUE THREAD

**Issue by RayYung** [FTBTeam/FTB-Presents-Direwolf20-1.12#53]
* **Modpack Version**: upgrade from 2.10 to 2.2.0

* **Issue**: When I upgrade, all TE machines "forgot" their orientation, tier and sides, some TE machines also lost the texture, some TD duct disconnected and many random connector in the wrong direction.

* **Link to Log or Crash File Paste**: (will come next)

* **Is it Repeatable?**: I removed and reinstalled the pack and still happen. Also try an old backup and repeat the same.

* **Mod/s Affected**: Thermal family

* **Known Fix**: I don't have any.

**Comment by RayYung** [comment]
Before upgrade:
![2018-08-20_12 02 51](https://user-images.githubusercontent.com/32179642/44358786-f8677180-a483-11e8-8665-774107eb38ef.png)

After:
![2018-08-20_14 20 52](https://user-images.githubusercontent.com/32179642/44358929-64e27080-a484-11e8-8d10-1d1b60715daa.png)
![2018-08-20_14 21 25](https://user-images.githubusercontent.com/32179642/44358932-67dd6100-a484-11e8-81c6-9a9f919893d6.png)

MC log: https://drive.google.com/file/d/1ZCxDyXn0xH3uTy7lVAoHCBUeluUGrxz-/view?usp=sharing
(too big for pastebin.....)

**Comment by RayYung** [comment]
Any one have suggestion for me? Should I just continue playing and trying to fix one by one ;
or roll back and wait and hope for a fix?

**Comment by RayYung** [comment]
I think I have to, just find yesterday the machines keep "forgot" after I
fix and save......


--
Raymond Yung
MCITP, MCTS, MCSA
RaymondY88@Gmail.com

**Comment by Kanzuke117** [comment]
I think I may have had the same issue on a multiplayer server, after updating from 2.0.0 to 2.2.0. 

Two chunks in particular were full of errors like the ones I can see in your screenshots, primarily affecting Refined Storage, Thermal Expansion, Draconic Evolution, and Chisel and Bits. Cables no longer seeming to be attached, chisel blocks missing, importers, exporters and crafters missrotated, etc.

My question would be, does everything still seem to `work`? Are materials still moving through the broken pipes? Are the crafters that look offline still able to produce their patterns?

What I observed was that the server was failing to send to the right information to the client for a lot of the contents in my two 'corrupted' chunks, so it all looked broken as hell, yet all my autocrafting still functioned, unlinked IO crystals were still draining power, etc. If I wrench everything back into working order, then walk away and back, it all appears broken again. Sound familiar?

**Comment by RayYung** [comment]
You are right, after I walk around and try to fix them. I found that there are more things corrupted then just Thermal. 
And yes, they keep going bad after I unload and reload...

**Comment by Kanzuke117** [comment]
Well, at least you have a backup from before the upgrade. If can, try updating mods one by one by copying the jars from a 2.2 install into a 2.1 install, and seeing if any one or combination of updates caused this issue. If you can find something, hopefully updating from 2.1 to a 2.3.0 pack could be the solution

**Comment by RayYung** [comment]
@SnowShock35 
SnowShock,

Care to share which Mod you think is causing the problem?
I will disable or update that Mod if that will let me continue to play..

Thanks,

**Comment by SnowShock35** [comment]
@RayYung 

I have absolutely no idea. It could be anything in all honesty.

**Comment by RayYung** [comment]
sigh.... Still broke after the new 2.3.0......

**Comment by SnowShock35** [comment]
It's highly likely that it was caused by world corruption. Really sorry that it's happened but I'm pretty sure I cannot do anything. 

**SnowShock35 ☃️**

***

> ![SnowShock35](https://cdn.snowshock35.com/avatar-rounded.png)
> ### SnowShock35
> @FTBTeam Pack Developer, Aspiring Full-Stack Engineer, 
> UoN Student, Mod & Modpack Developer, Casual Gamer,
> Nerd, Tea Drinker, Made in the UK 🏳️‍🌈

**Comment by RayYung** [comment]
I felt that coming......  :'( 
Any suggestion on any tools or mod that can help me move a wall of machines to a different place.

**Comment by SnowShock35** [comment]
Not that I'm aware of in this pack. Perhaps funky locomotion could help but the easiest thing to do with machines is getting a strong box for each machine setup you have and fill it with all the bits (as well as take screenshots of what it looked like) and to rebuild it. 

Also, there's likely a backup prior to your machines becoming broken. Try reverting to that and playing it.

**Comment by RayYung** [comment]
I still have the working save, I was still playing 2.1.0. however, I wanna play with new mods and a new version. 
I will try funky locomotion so I don't have to rebuild everything..

Thanks,

**Comment by SnowShock35** [comment]
Try the version that was all fine in 2.3.0 - see what that does too.

**Comment by RayYung** [comment]
That's what I did yesterday. (load a working 2.10 world into a brand new 2.3.0 instance.)
It acts the same as 2.2, breaking a lot of machines...

**Comment by SnowShock35** [comment]
Oh right 😕
